csharp/ql/lib/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -2212,6 +2212,15 @@ predicate allowParameterReturnInSelf(ParameterNode p) {
   FlowSummaryImpl::Private::summaryAllowParameterReturnInSelf(p)
 }
 
+/**
+ * A module importing the modules that provide synthetic field declarations,
+ * ensuring that they are visible to the taint tracking / data flow library.
+ */
+private module SyntheticFields {
+  private import semmle.code.csharp.frameworks.system.threading.Tasks
+  private import semmle.code.csharp.frameworks.system.runtime.CompilerServices
```

Comment:
This files only had code deleted, so why is this needed now and not before (perhaps it was)?

Reply:
This is to ensure the bi-directional import for synthetic fields - this is implicitly handled by the bi-directional import in the *Frameworks* module in `ExternalFlow.qll`.
It turns out that the massive gordian knot that is the *Frameworks* module in ExternalFlow can't *just* be deleted (this needs to be a follow up, if we want to do that). I have document the this on: https://github.com/github/codeql-csharp-team/issues/298

Reply:
Added an import for a similar reason in `TaintTrackingPrivate.qll`